pkg/operator/operator.go

```diff
@@ -202,6 +208,81 @@ func New(logger logr.Logger, clientConfig *rest.Config, opts Options) (*Operator
 	if err != nil {
 		return nil, fmt.Errorf("invalid port: %w", err)
 	}
+
+	watchObjects := map[client.Object]cache.ByObject{
+		&corev1.Pod{}: {
+			Field: fields.SelectorFromSet(fields.Set{"metadata.namespace": opts.OperatorNamespace}),
+		},
+		&monitoringv1.PodMonitoring{}: {
+			Field: fields.Everything(),
+		},
+		&monitoringv1.ClusterPodMonitoring{}: {
+			Field: fields.Everything(),
+		},
+		&monitoringv1.ClusterNodeMonitoring{}: {
+			Field: fields.Everything(),
+		},
+		&monitoringv1.GlobalRules{}: {
+			Field: fields.Everything(),
+		},
+		&monitoringv1.ClusterRules{}: {
+			Field: fields.Everything(),
+		},
+		&monitoringv1.Rules{}: {
+			Field: fields.Everything(),
+		},
+		&corev1.Secret{}: {
+			Namespaces: map[string]cache.Config{
+				opts.OperatorNamespace: {},
+				opts.PublicNamespace:   {},
+			},
+		},
+		&monitoringv1.OperatorConfig{}: {
+			Field: fields.SelectorFromSet(fields.Set{"metadata.namespace": opts.PublicNamespace}),
+		},
+		&corev1.Service{}: {
+			Field: fields.SelectorFromSet(fields.Set{
+				"metadata.namespace": opts.OperatorNamespace,
+				"metadata.name":      NameAlertmanager,
+			}),
+		},
+		&corev1.ConfigMap{}: {
+			Field: fields.SelectorFromSet(fields.Set{"metadata.namespace": opts.OperatorNamespace}),
+		},
+		&appsv1.DaemonSet{}: {
+			Field: fields.SelectorFromSet(fields.Set{
+				"metadata.namespace": opts.OperatorNamespace,
+				"metadata.name":      NameCollector,
+			}),
+		},
+		&appsv1.Deployment{}: {
+			Field: fields.SelectorFromSet(fields.Set{
+				"metadata.namespace": opts.OperatorNamespace,
+				"metadata.name":      NameRuleEvaluator,
+			}),
+		},
+	}
+
+	// In GKE, installing Vertical Pod Autoscaling requires a cluster restart, and therefore it also results in an operator restart.
+
+	// In other environments, the operator may need to be restarted to enable VPA to run the following check again and watch for the objects.
+	var vpaAvailable bool
+	clientset, err := apiextensions.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, fmt.Errorf("create clientset: %w", err)
+	}
+	if _, err := clientset.ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(), "verticalpodautoscalers.autoscaling.k8s.io", metav1.GetOptions{}); err != nil {
+		logger.Info("vertical pod autoscaling is not available")
```

Comment:
Can we mention in this log line, what operator is doing with this fact (same below?) e.g.

```suggestion
		logger.Info("vertical pod autoscaling is not available, vpa.enabled config option will not work")
```